### PR TITLE
Feature/custom mounts

### PR DESCRIPTION
**Pull request by patatman**
#### Summary

This adds the possibility to add custom mounts inside the Daemonset.
For example for custom CA's.
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-helm/issues/375

### CODE REVIEW

**Review by phoinixgrr — COMMENTED** [review]
Happy merging this after we fix some NL characters introduced in the templating

Rendered Daemonset manifest: 
```yaml
kind: DaemonSet
metadata:
  name: release-name-mattermost-rtcd
  labels:
    helm.sh/chart: mattermost-rtcd-1.1.1
    app.kubernetes.io/name: mattermost-rtcd
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "latest"
    app.kubernetes.io/managed-by: Helm
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: mattermost-rtcd
      app.kubernetes.io/instance: release-name
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mattermost-rtcd
        app.kubernetes.io/instance: release-name
    spec:
      dnsPolicy: ClusterFirstWithHostNet
      hostNetwork: true
      serviceAccountName: release-name-mattermost-rtcd
      terminationGracePeriodSeconds: 18000
      securityContext:
        {}
      containers:
        - name: mattermost-rtcd
          securityContext:
            {}
          image: "mattermost/rtcd:latest"
          imagePullPolicy: IfNotPresent
          volumeMounts:

          - mountPath: /host/var/log
            name: varlog
            readOnly: true
          ports:
            - name: api
              containerPort: 8045
              hostPort: 8045
              protocol: TCP
            - name: rtc
              containerPort: 8443
              hostPort: 8443
              protocol: UDP
          livenessProbe:
            httpGet:
              path: /version
              port: api
          readinessProbe:
            httpGet:
              path: /version
              port: api
          resources:
            {}
      volumes:

        - hostPath:
            path: /var/log
          name: varlog
```

**Comment by patatman** [comment]
@phoinixgrr I've added a NL to the EOF, is this what you ment?

**Comment by phoinixgrr** [comment]
> @phoinixgrr I've added a NL to the EOF, is this what you ment?

Nice 👍  
Thank you @patatman 

I'am also referring to the two NL characters introduced in the daemonset template, under `volumeMounts` and `volumes` directives.

**Comment by patatman** [comment]
Ah right! I didn't spot that my bad.
Not sure how to trim those? Do you have any idea? 
I'll go and try some things and try to make them go away.

**Comment by patatman** [comment]
@phoinixgrr Already fixed it, looks like I needed to remove the NL after the yaml template

File: charts/mattermost-rtcd/templates/daemonset.yaml [hunk 1]
```diff
@@ -81,3 +84,6 @@ spec:
       dnsConfig:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.daemonset.extraVolumes }}
+      volumes: {{ toYaml .Values.daemonset.extraVolumes | nindent 8 }}
```
Comment (patatman) [inline charts/mattermost-rtcd/templates/daemonset.yaml:88]: I added a NL for readability here, but it messed up the generated YAML. 
Putting these 2 on 1 line seems to fix this.

Comment (phoinixgrr) [reply charts/mattermost-rtcd/templates/daemonset.yaml:88]: Let's also correctly indent `volumeMounts`:, to align with the rest of the template 


```
          volumeMounts:
          - mountPath: /host/var/log
            name: varlog
            readOnly: true
```

to 
```
          volumeMounts:
            - mountPath: /host/var/log
              name: varlog
              readOnly: true
```

Comment (patatman) [reply charts/mattermost-rtcd/templates/daemonset.yaml:88]: Done!

**Review by phoinixgrr — APPROVED** [review]

**Comment by phoinixgrr** [comment]
Triggering tests

**Comment by phoinixgrr** [comment]
cc: @spirosoik  @stylianosrigas  @stafot  @angeloskyratzakos 

Any objections merging this?

**Review by stafot — APPROVED** [review]
LGTM

**Review by spirosoik — APPROVED** [review]
LGTM. @patatman thanks for the contribution 💪

**Comment by phoinixgrr** [comment]
Merged 
@patatman thanks for your contribution 🎖️ 🏅 🥇

**Pull request merged**